pooch/tests/test_core.py

```diff
@@ -517,6 +518,7 @@ def test_check_availability_on_ftp(ftpserver):
         assert not pup.is_available("doesnot_exist.zip", downloader=downloader)
 
 
+@pytest.mark.network
```

Comment:
This test should not require network connection. The only reason it does is because it tests if everything works without the custom downloader, but this is already covered by `test_check_availability`. So I think we should remove the `network` mark and update the test to something like:
```python
def test_check_availability_invalid_downloader():
    "Should raise an exception if the downloader doesn't support this"

    def downloader(url, output, pooch):  # pylint: disable=unused-argument
        "A downloader that doesn't support check_only"
        return None

    pup = Pooch(path=DATA_DIR, base_url=BASEURL, registry=REGISTRY)
    msg = "does not support availability checks."
    with pytest.raises(NotImplementedError, match=msg):
        pup.is_available("tiny-data.txt", downloader=downloader)
```

Reply:
Currently the test fails with
```python
E       socket.gaierror: [Errno -3] Temporary failure in name resolution
```

I'm fine with changing the test in favor of marking it.

Reply:
Good to know. Does my new proposal pass on your side?

Reply:
Yes, it does. Thank you very much.